assets/tests/api.test.ts

```diff
@@ -230,6 +232,97 @@ describe("checkedApiCall", () => {
   })
 })
 
+describe("apiCallWithError", () => {
+  let browserReloadSpy: SpyInstance
+
+  beforeEach(() => {
+    browserReloadSpy = jest
+      .spyOn(browser, "reload")
+      .mockImplementation(() => {})
+  })
+
+  afterAll(() => {
+    browserReloadSpy.mockRestore()
+  })
+
+  test("returns parsed data", async () => {
+    mockFetch(200, { data: "raw" })
+
+    const parse = jest.fn(() => "parsed")
+
+    return apiCallWithError({
+      url: "/",
+      dataStruct: string(),
+      parser: parse,
+    }).then((parsed) => {
+      expect(parse).toHaveBeenCalledWith("raw")
+      expect(parsed).toEqual(ok("parsed"))
+    })
```

Comment:
```suggestion
    await expect(apiCallWithError({
      url: "/",
      dataStruct: string(),
      parser: parse,
    })).resolves.toEqual(ok("parsed"))
    expect(parse).toHaveBeenCalledWith("raw")
```

Reply:
Have a look at d73dfc27 - I also removed one test because I realized we no longer have a default return value.